File: component-plugin/src/main/groovy/com/plugin/component/anno/AutoInjectComponent.java
package com.plugin.component.anno;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 自动注入组件实现
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.CLASS)
public @interface AutoInjectComponent {
    Class[] impl();
}

File: component-core/src/main/java/com/plugin/component/anno/AutoInjectComponent.kt
package com.plugin.component.anno

import kotlin.reflect.KClass

/**
 * 自动注入组件实现
 */
@Target(AnnotationTarget.CLASS)
@kotlin.annotation.Retention(AnnotationRetention.BINARY)
annotation class AutoInjectComponent(val impl: Array<KClass<*>>)
